src/components/Dashboard/Dashboard.tsx

```diff
@@ -11,81 +14,213 @@ import {
 import { DashboardTheme } from "./DashboardTheme";
+import { Sidebar } from "./Sidebar";
 import { Toolbar } from "../Toolbar/Toolbar";
+import { TCacheKey } from "../../types/types";
 
 /**
  * The Dashboard component summarizes disparate types of information into a series of widgets.
  * Designs for this component are available in the [Dashboard page of the Microsoft Teams UI Kit](https://www.figma.com/file/EOsbapNvZgEwcA1mShswfh/Microsoft-Teams-UI-Kit-Community?node-id=3789%3A3890).
  * @public
  */
 export interface IDashboard {
+  /**
+   * The widgets to make available in this Dashboard.
+   */
   widgets: IWidget[];
+  /**
+   * Any initial preferences that should be set for this user, in case you save users’ preferences remotely.
+   */
+  preferences?: IDashboardPreferences;
+  /**
+   * Set this `cacheKey` to save user’s preferences on their local clients.
+   */
+  cacheKey?: TCacheKey;
+  /**
+   * A Dashboard will emit onInteraction payloads when the user updates any preferences.
+   */
+  onInteraction?: (interaction: TDashboardInteraction) => void;
 }
 
-const toolbarConfig = {
-  actionGroups: {
-    h1: {
-      edit: { title: "Edit dashboard", icon: "Edit" },
-    },
-  },
-  filters: [],
-  find: false,
-};
+/**
+ * A user’s preferences for the particular Dashboard component. For a given Dashboard instance,
+ * a user may set certain widgets to be shown or hidden.
+ * @public
+ */
+export interface IDashboardPreferences {
+  // widgetOrder: string[];
+  widgetSettings: {
+    [widgetKey: string]: {
+      display: boolean;
+    };
+  };
+}
 
 /**
+ * A Dashboard will emit onInteraction payloads when the user updates any preferences.
  * @public
  */
-export function Dashboard({ widgets }: IDashboard) {
+export type TDashboardInteraction = IDashboardInteractionUpdatePreferences;
+
+/**
+ * The preferences update payload carries the preferences the developer should store for the user,
+ * if appropriate.
+ * @public
+ */
+export interface IDashboardInteractionUpdatePreferences {
+  event: "update";
+  target: "preferences";
+  preferences: IDashboardPreferences;
+}
+
+const emptyPrefs = { widgetSettings: {} };
+
+/*
+ * This method returns the same stored preferences for any Dashboard.
+ */
+const getStoredPrefs = (cacheKey: TCacheKey) =>
+  cacheKey
+    ? (() => {
+        const storedPrefs = window.localStorage.getItem(cacheKey);
+        return storedPrefs ? JSON.parse(storedPrefs) : false;
+      })() || emptyPrefs
+    : (emptyPrefs as IDashboardPreferences);
+
+/**
+ * @public
+ */
+export function Dashboard({
+  widgets,
+  preferences,
+  cacheKey,
+  onInteraction,
+}: IDashboard) {
+  const [sidebarOpen, setSidebarOpen] = useState<boolean>(false);
+  const closeSidebar = () => setSidebarOpen(false);
+
+  const localStorageKey = cacheKey
+    ? `@fluentui/react-teams__${cacheKey}`
+    : undefined;
+
+  const [preferencesState, setPreferencesState] = useState<
+    IDashboardPreferences
+  >(
+    (() => {
```

Comment:
Should probably pass a function to `useState` here (i.e. not invoke your IIFE). Right now each render will read from local storage (if `preferences` is undefined) and perform a deep clone.

Reply:
Ah right, I must have forgotten to make that last refinement, thank you!

Reply:
Ah right, thank you!